app/test/tool/backfill/backfill_packageversions_test.dart

```diff
@@ -97,5 +101,51 @@ void main() {
         expect(readme2.textContent, readme.textContent);
       },
     );
+
+    testWithProfile(
+      'Update pubspec in PackageVersion and in PackageVersionAsset',
+      fn: () async {
+        final pv =
+            (await packageBackend.lookupPackageVersion('oxygen', '1.2.0'))!;
+        final asset = (await packageBackend.lookupPackageVersionAsset(
+          'oxygen',
+          '1.2.0',
+          AssetKind.pubspec,
+        ))!;
+        final originalPvContent = pv.pubspec!.jsonString;
+        final originalAssetContent = asset.textContent;
+
+        pv.pubspec = Pubspec.fromYaml('name: x');
+        asset.textContent = 'name: x\nversion: 1,2,3\n';
+        await dbService.commit(inserts: [pv, asset]);
+
+        final stat = await backfillAllVersionsOfPackage(
+          'oxygen',
+          archiveResolver: _archive,
+        );
+
+        expect(
+          stat.toJson(),
+          {
+            'versionCount': 3,
+            'pvCount': 1,
+            'pvInfoCount': 0,
+            'pvAssetUpdatedCount': 1,
+            'pvAssetDeletedCount': 0
+          },
+        );
```

Comment:
Isn't it a little fragile to test on this?

Shouldn't we only make assertions over `pvCount`, or maybe just not make an assertion here?